frontend/src/metabase/App.jsx

```diff
@@ -61,31 +83,128 @@ export default class App extends Component {
     this.setState({ errorInfo });
   }
 
+  isAdminApp = () => {
+    const { pathname } = this.props.location;
+    return pathname.startsWith("/admin/");
+  };
+
   hasNavbar = () => {
+    const {
+      currentUser,
+      isEditingDashboard,
+      location: { pathname },
+    } = this.props;
+    if (!currentUser || IFRAMED || isEditingDashboard) {
+      return false;
+    }
+    return !PATHS_WITHOUT_NAVBAR.some(pattern => pattern.test(pathname));
+  };
+
+  hasAppBar = () => {
     const {
       currentUser,
       location: { pathname },
+      isEditingDashboard,
     } = this.props;
-    if (!currentUser || IFRAMED) {
+    if (!currentUser || IFRAMED || this.isAdminApp() || isEditingDashboard) {
       return false;
     }
     return !PATHS_WITHOUT_NAVBAR.some(pattern => pattern.test(pathname));
   };
 
+  closeModal = () => {
+    this.setState({ modal: null });
+  };
+
+  setModal = modal => {
+    this.setState({ modal });
+    if (this._newPopover) {
+      this._newPopover.close();
+    }
+  };
+
+  renderModalContent() {
+    const { modal } = this.state;
+    const { onChangeLocation } = this.props;
+
+    switch (modal) {
+      case MODAL_NEW_COLLECTION:
+        return (
+          <CollectionCreate
+            onClose={() => this.setState({ modal: null })}
+            onSaved={collection => {
+              this.setState({ modal: null });
+              onChangeLocation(Urls.collection(collection));
+            }}
+          />
+        );
+      case MODAL_NEW_DASHBOARD:
+        return (
+          <CreateDashboardModal
+            onClose={() => this.setState({ modal: null })}
+          />
+        );
+      default:
+        return null;
+    }
+  }
+
+  renderModal = () => {
+    const { modal } = this.state;
+
+    if (modal) {
+      return (
+        <Modal onClose={this.closeModal}>{this.renderModalContent()}</Modal>
+      );
+    } else {
+      return null;
+    }
+  };
+
   render() {
-    const { children, location, errorPage } = this.props;
+    const {
+      isSidebarOpen,
+      children,
+      location,
+      errorPage,
+      onChangeLocation,
+      toggleNavbar,
+    } = this.props;
     const { errorInfo } = this.state;
-
+    const hasAppBar = this.hasAppBar();
     return (
       <ScrollToTop>
-        <div className="relative">
-          {this.hasNavbar() && <Navbar location={location} />}
-          {errorPage ? getErrorComponent(errorPage) : children}
-          <UndoListing />
-          <StatusListing />
-        </div>
+        {errorPage ? (
+          getErrorComponent(errorPage)
+        ) : (
+          <>
+            {hasAppBar && (
+              <AppBar
```

Comment:
Can we move state from `App` component to redux and connect `AppBar` and another component to it? I believe that we should `App` is small is possible.